framework/src/st_things/things_stack/src/stack/things_stack.c

```diff
@@ -0,0 +1,1149 @@
+
+	for (; retry_count < 3; retry_count++) {
+		result = wifi_manager_connect_ap(connect_config);
+
+		if (result == WIFI_MANAGER_SUCCESS) {
+			break;
+		} else {
+			THINGS_LOG_ERROR(THINGS_ERROR, TAG, "Failed to connect WiFi");
+		}
+	}
+
+	if (retry_count == 3) {
+		THINGS_LOG_ERROR(THINGS_ERROR, TAG, "Failed to connect WiFi 3 times");
+		return false;
+	}
+
+	if (connect_config != NULL) {
+		save_acces_point_info(connect_config);
+	}
+
+	return true;
+}
+
+INT8 things_wifi_connection_cb(access_point_info_s *p_info, char *p_cmd_id)
+{
+	THINGS_LOG_D(THINGS_INFO, TAG, "T%d --> %s", getpid(), __FUNCTION__);
+
+	if (p_info == NULL) {
+		THINGS_LOG_D(THINGS_INFO, TAG, "Invalid params");
+		return -1;
+	}
+
+	wifi_manager_ap_config_s connect_config;
+
+	strncpy(connect_config.ssid, p_info->e_ssid, sizeof(connect_config.ssid));
+	connect_config.ssid_length = strlen(connect_config.ssid);
+	strncpy(connect_config.passphrase, p_info->security_key, sizeof(connect_config.passphrase));
+
+	connect_config.passphrase_length = strlen(connect_config.passphrase);
+
+	THINGS_LOG_D(THINGS_INFO, TAG, "[%s] ssid : %s", __FUNCTION__, connect_config.ssid);
+	THINGS_LOG_D(THINGS_INFO, TAG, "[%s] passphrase : %s", __FUNCTION__, connect_config.passphrase);
+
+	// set auth type
+	if (strcmp(p_info->auth_type, "WEP") == 0) {
+		connect_config.ap_auth_type = WIFI_MANAGER_AUTH_WEP_SHARED;
+	} else if (strcmp(p_info->auth_type, "WPA-PSK") == 0) {
+		connect_config.ap_auth_type = WIFI_MANAGER_AUTH_WPA_PSK;
+	} else if (strcmp(p_info->auth_type, "WPA2-PSK") == 0) {
+		connect_config.ap_auth_type = WIFI_MANAGER_AUTH_WPA2_PSK;
+	}
+	// set encryption crypto type
+	if (strcmp(p_info->enc_type, "WEP-64") == 0) {
+		connect_config.ap_crypto_type = WIFI_MANAGER_CRYPTO_WEP_64;
+	} else if (strcmp(p_info->enc_type, "WEP-128") == 0) {
+		connect_config.ap_crypto_type = WIFI_MANAGER_CRYPTO_WEP_128;
+	} else if (strcmp(p_info->enc_type, "TKIP") == 0) {
+		connect_config.ap_crypto_type = WIFI_MANAGER_CRYPTO_TKIP;
+	} else if (strcmp(p_info->enc_type, "AES") == 0) {
+		connect_config.ap_crypto_type = WIFI_MANAGER_CRYPTO_AES;
+	} else if (strcmp(p_info->enc_type, "TKIP_AES") == 0) {
+		connect_config.ap_crypto_type = WIFI_MANAGER_CRYPTO_TKIP_AND_AES;
+	}
+
+	int ret = things_handle_stop_soft_ap(&connect_config);
+	if (ret != 1) {
+		return -1;
+	}
+
+	return 1;
+}
+
+access_point_info_s *create_access_info()
+{
+	access_point_info_s *info = (access_point_info_s *) things_malloc(sizeof(access_point_info_s));
+	memset(info->e_ssid, 0, MAX_ESSID);
+	memset(info->security_key, 0, MAX_SECUIRTYKEY);
+	memset(info->auth_type, 0, MAX_TYPE_AUTH);
+	memset(info->enc_type, 0, MAX_TYPE_ENC);
+	memset(info->channel, 0, MAX_CHANNEL);
+	memset(info->signal_level, 0, MAX_LEVEL_SIGNAL);
+	memset(info->bss_id, 0, MAX_BSSID);
+	return info;
+}
+
+int things_wifi_search_cb(access_point_info_s *** p_info, int *p_count)
+{
+	THINGS_LOG_V(THINGS_INFO, TAG, THINGS_FUNC_ENTRY);
+#if 0							// TO DO
```

Comment:
Please, remove the unused code.

Reply:
It will be implemented later. Should I erase it anyway?

Reply:
I think you can keep this in private.